src/mutator.cc

```diff
@@ -457,7 +457,45 @@ void Mutator::Mutate(Message* message, size_t size_increase_hint) {
   InitializeAndTrim(message, kMaxInitializeDepth);
   assert(!keep_initialized_ || message->IsInitialized());
 
-  if (post_process_) post_process_(message, random_());
+  ApplyPostProcessing(message);
+}
+
+void Mutator::RegisterMutator(
+    const protobuf::Descriptor* desc,
+    std::function<void(protobuf::Message* msg)> callback) {
+  custom_mutations_.emplace(desc, callback);
+}
+
+void Mutator::ApplyPostProcessing(Message* message) {
+  if (post_process_) {
```

Comment:
we don't want to call post_process_ for all nested messages
also type is not going to match.

I proposed to remove post_process_ and just make RegisterPostProcessorImpl to register for root message type